resources/js/Shared/Modules/Addresses.vue

```diff
@@ -1,3 +1,142 @@
+
+const props = defineProps({
+  layoutData: Object,
+  data: Object,
+});
+
+const loadingState = ref('');
+const createAddressModalShown = ref(false);
+const inactiveAddressesShown = ref(false);
+const localActiveAddresses = ref([]);
+const localInactiveAddresses = ref([]);
+const editedAddressId = ref(0);
+const choiceChooseExisting = ref(true);
+
+const form = useForm({
+  existing_address: false,
+  existing_address_id: 0,
+  type: '',
+  address_type_id: 0,
+  is_past_address: false,
+  line_1: '',
+  line_2: '',
+  city: '',
+  province: '',
+  postal_code: '',
+  country: '',
+  errors: [],
+});
+
+onMounted(() => {
+  localActiveAddresses.value = props.data.active_addresses;
+  localInactiveAddresses.value = props.data.inactive_addresses;
+  choiceChooseExisting.value = true;
+});
+
+const showCreateAddressModal = () => {
+  form.errors = [];
+  form.is_past_address = false;
+  form.address_type_id = 0;
+  form.line_1 = '';
+  form.line_2 = '';
+  form.city = '';
+  form.province = '';
+  form.postal_code = '';
+  form.country = '';
+  createAddressModalShown.value = true;
+};
+
+const toggleInactiveAdresses = () => {
+  inactiveAddressesShown.value = !inactiveAddressesShown.value;
+};
+
+const showEditAddressModal = (address) => {
+  editedAddressId.value = address.id;
+  form.errors = [];
+  form.is_past_address = address.is_past_address;
+  form.address_type_id = address.type ? address.type.id : 0;
+  form.line_1 = address.line_1;
+  form.line_2 = address.line_2;
+  form.city = address.city;
+  form.province = address.province;
+  form.postal_code = address.postal_code;
+  form.country = address.country;
+};
+
+const submit = () => {
+  loadingState.value = 'loading';
+  form.existing_address = choiceChooseExisting.value;
+
+  axios
+    .post(props.data.url.store, form)
+    .then((response) => {
+      if (form.is_past_address) {
+        localInactiveAddresses.value.unshift(response.data.data);
+      } else {
+        localActiveAddresses.value.unshift(response.data.data);
+      }
+
+      loadingState.value = '';
+      createAddressModalShown.value = false;
+    })
+    .catch((error) => {
+      loadingState.value = '';
+      form.errors = error.response.data;
+    });
+};
+
+const update = (address) => {
+  loadingState.value = 'loading';
+
+  axios
+    .put(address.url.update, form)
+    .then((response) => {
+      loadingState.value = '';
+
+      if (form.is_past_address) {
+        localInactiveAddresses.value[localInactiveAddresses.value.findIndex((x) => x.id === address.id)] =
+          response.data.data;
+      } else {
+        localActiveAddresses.value[localActiveAddresses.value.findIndex((x) => x.id === address.id)] =
+          response.data.data;
+      }
+      editedAddressId.value = 0;
+    })
+    .catch((error) => {
+      loadingState.value = '';
+      form.errors = error.response.data;
+    });
+};
+
+const destroy = (address) => {
+  if (confirm('Are you sure? This can not be undone.')) {
```

Comment:
Can you use the `ConfirmationModal` module instead?

Reply:
Done